guide/backend/src/main/scala/io/udash/web/server/ApplicationServer.scala

```diff
@@ -49,53 +55,30 @@ class ApplicationServer(val port: Int, homepageResourceBase: String, guideResour
       })
 
       val framework = new DefaultAtmosphereFramework(config, exceptionsRegistry = GuideExceptions.registry)
-      val atmosphereHolder = new ServletHolder(new RpcServlet(framework))
-      atmosphereHolder.setAsyncSupported(true)
-      atmosphereHolder
+      new ServletHolder(new RpcServlet(framework))
     }
-    ctx.addServlet(atmosphereHolder, "/atm/*")
+    contextHandler.addServlet(atmosphereHolder, "/atm/*")
 
     //required for org.atmosphere.container.JSR356AsyncSupport
-    JavaxWebSocketServletContainerInitializer.configure(ctx, null)
+    JavaxWebSocketServletContainerInitializer.configure(contextHandler, null)
 
-    val restHolder = new ServletHolder(
-      RestServlet[MainServerREST](new ExposedRestInterfaces)
-    )
-    restHolder.setAsyncSupported(true)
-    ctx.addServlet(restHolder, "/rest_api/*")
-    ctx
-  }
+    contextHandler.addServlet(new ServletHolder(RestServlet[MainServerREST](new ExposedRestInterfaces)), "/rest_api/*")
 
-  private val contexts = new ContextHandlerCollection
-  contexts.setHandlers(Array(homepage, guide))
-
-  private val rewriteHandler = {
-    import org.eclipse.jetty.rewrite.handler.RewriteRegexRule
-    val rewrite = new org.eclipse.jetty.rewrite.handler.RewriteHandler()
-    rewrite.setRewriteRequestURI(true)
-    rewrite.setRewritePathInfo(false)
-
-    val spaRewrite = new RewriteRegexRule
-    spaRewrite.setRegex("^/(?!assets|scripts|styles|atm|rest_api)(.*/?)*$")
-    spaRewrite.setReplacement("/")
-    rewrite.addRule(spaRewrite)
-    rewrite.setHandler(contexts)
-    rewrite
+    new GzipHandler(contextHandler.get())
   }
 
-  server.setHandler(rewriteHandler)
-
-  private def createContextHandler(hosts: Array[String]): ServletContextHandler = {
-    val context = new ServletContextHandler(ServletContextHandler.SESSIONS)
-    context.insertHandler(new GzipHandler)
-    context.setVirtualHosts(hosts)
-    context
+  server.setHandler(
+    new RewriteHandler(new ContextHandlerCollection().setup(_.setHandlers(homepage, guide)))
+      .setup(_.addRule(new RewriteRegexRule("^/(?!assets|scripts|styles|atm|rest_api)(.*/?)*$", "/")))
+  )
+
+  private def createContextHandler(hosts: Array[String], resourceBase: String): ServletContextHandler = {
```

Comment:
I would consider adding normalization to resourceBase as jetty 12 changed implementation of SymlinkAllowedResourceAliasChecker and it no longer allows resources with /../ in path resulting in 404 errors.